pkg/interface/chat/src/js/components/chat.js

```diff
@@ -84,26 +89,32 @@ export class ChatScreen extends Component {
   askForMessages() {
     const { props, state } = this;
 
+    if (props.envelopes.length === 0) {
+      setTimeout(() => {
+        this.askForMessages();
+      }, 500);
```

Comment:
please explain

Reply:
In the case that you have refreshed the browser or directed the browser directly to a specific chatroom, you haven’t set up your subscriptions / received envelopes yet. Thus, props.envelopes.length === 0. Thus, to successfully askForMessages, we set a timeout to call askForMessages again once we’ve loaded the envelopes properly. The “chatroom has no messages ever and we call this method twice a second” edge case seems like a reasonable tradeoff.